librz/analysis/esil/esil_trace.c

```diff
@@ -126,8 +292,8 @@ static int trace_hook_reg_read(RzAnalysisEsil *esil, const char *name, ut64 *res
 	if (ret) {
 		ut64 val = *res;
 		//eprintf ("[ESIL] REG READ %s 0x%08"PFMT64x"\n", name, val);
-		sdb_array_add(DB, KEY("reg.read"), name, 0);
-		sdb_num_set(DB, KEYREG("reg.read", name), val, 0);
+		rz_analysis_esil_trace_db_array_add(DB, KEY("reg.read"), name);
```

Comment:
I wonder, what about creating 4 different hashtables for reading/writing memory, reading/writing registers? It should speed up the access as well. cc @pelijah @ret2libc